### ISSUE TITLE
Airship test and some improvement suggestions

### ISSUE THREAD

**Issue by by003** [vcmi-mods/horn-of-the-abyss#107]
![image](https://github.com/vcmi-mods/horn-of-the-abyss/assets/127420617/1985ed42-947f-4b8e-b542-0f34a87dc2ce)

I use core:boat to create airship, with this codes:
```
{
	"core:boat" : {
		"types" : {
			"airshipFactory" : {
				"removable": true,
				"rmg" : {
				},
				"layer" : "air",
				// If set to true, it is possible to attack wandering monsters from boat
				"onboardAssaultAllowed" : false,
				// If set to true, it is possible to visit object (e.g. pick up resources) from boat
				"onboardVisitAllowed" : false,
				"actualAnimation" : "factory/map/objects/AAir00.def",
				"overlayAnimation" : "ABM03_.def",
				"flagAnimations" : [
				"factory/map/objects/AAirF00L.def",
				"factory/map/objects/AAirF00G.def",
				"factory/map/objects/AAirF00R.def",
				"factory/map/objects/AAirF00D.def",
				"factory/map/objects/AAirF00B.def",
				"factory/map/objects/AAirF00P.def", 
				"factory/map/objects/AAirF00W.def",
				"factory/map/objects/AAirF00K.def"],
				"templates" : {
					"default" : {
						"animation" : "factory/map/objects/AAir00_E.def",
						"visitableFrom" : [ "+++", "+-+", "+++" ],
						"mask" : [ "VVV","VVV","VVV", "VAV" ]
					}
				}
			}
		}
	}
}
```
Now there are some problems:
1. Hero cannot enter the water directly, but can directly go ashore.
![image](https://github.com/vcmi-mods/horn-of-the-abyss/assets/127420617/887d2fd2-cd3d-496b-a3c6-eb3921165feb)
2. If hero needs to enter the water, a boat is needed to become a ‘springboard’, if there is no boat, this hero can never enter the water.
![image](https://github.com/vcmi-mods/horn-of-the-abyss/assets/127420617/3bd698c1-57ee-4c93-99f4-14043d71bac6)
3. This hero can attack others or pick theresources, depends on "onboardAssaultAllowed" and "onboardVisitAllowed" are set on 'true'.
![image](https://github.com/vcmi-mods/horn-of-the-abyss/assets/127420617/ea3e02b4-6a4a-4090-9805-ab74983a0480)
4. This hero CANNOT get off the ship, need a button to help him.

**Comment by GeorgeK1ng** [comment]
@by003 Any news on AirShip?

**Comment by by003** [comment]
> Any news on AirShip?

Not yet. I think need developers' help.

Maybe ask for @Laserlicht ?

**Comment by Laserlicht** [comment]
> Maybe ask for [@Laserlicht](https://github.com/Laserlicht) ?

Not really interested in it yet. Because it's related to some parts of code I've never worked on.

**Comment by by003** [comment]
https://github.com/vcmi/vcmi/issues/4051
Same here, if completed, just finish this issue and that one in vcmi part.

**Comment by GeorgeK1ng** [comment]
Have you tried it from https://github.com/vcmi/vcmi/pull/6464? Also not sure if this will be part of 1.7.x

**Comment by by003** [comment]
@GeorgeK1ng @Karyoplasma 

Some Chinese players downloaded maps containing airships to play, but found they couldn't disembark... but that's still a problem with version 1.7.1.

Thus, can we fully utilize the airship functionality now? Do we need to add anything new, such as a UI, to HotA's mod pack to support it? In which PR is the version with complete airship functionality currently located? Or should we wait until after the merge for testing?

**Comment by Karyoplasma** [comment]
i dunno, didn't keep track of airship, sorry :(

maybe @mrhaandi knows more?

**Comment by GeorgeK1ng** [comment]
Proper engine support is ready in https://github.com/vcmi/vcmi/pull/6464 but currently target to develop branch. Before VCMI 1.7 release it was only WIP. Now it looks almost ready - have not found other issues.

I think Its question for @IvanSavenko if possible to retarget to beta for 1.7.2 or 1.7.3. Would be great to cover AirShip support.

I know there Is animation issue, but nothing more. New "Anchor" button can be added then to VCMI Extras mod to be able to disembark.

**Comment by Laserlicht** [comment]
Yes. In general would be great to have it in 1.7.X as it adds many HotA related stuff.
But the question here is: Is it safe to do? If it was tested well enough: Why not?

**Comment by GeorgeK1ng** [comment]
@by003 Can you please test AirShip too? https://github.com/vcmi/vcmi/pull/6464#issuecomment-3839616339 Artifacts should be ready and it's based on beta branch (VCMI 1.7.2). So can be easily tested with latest HotA mod. Only require AirShip files from https://github.com/vcmi/vcmi/pull/6464#issue-3738415252

**Comment by by003** [comment]
Sure. I will test it later, now it is so late here.

**Comment by by003** [comment]
Most of the airship issues have been resolved.

A few remain:
1. The airship dock recruitment interface should be static; currently, it only shows an animation of rising and falling.
<img width="339" height="210" alt="Image" src="https://github.com/user-attachments/assets/1da0499f-4362-4f81-a305-b3def66dccb5" />

2. The rising and falling animation loops continuously during flight, which is visually unappealing.

https://github.com/user-attachments/assets/6170ea12-2d6a-43f2-9cbc-e21228e01ada

3. In HotA's setting, airships cannot attack any neutral forces or enemies, but it seems that battles will occur when they get close.
4. Airships should not play the sound of horses' hooves when sailing on land.

@GeorgeK1ng @Karyoplasma @Laserlicht
